pkg/controller/change/network/controller.go

```diff
@@ -155,23 +153,30 @@ func (r *Reconciler) createDeviceChanges(networkChange *networktypes.NetworkChan
 
 // canApplyChange returns a bool indicating whether the change can be applied
 func (r *Reconciler) canApplyChange(change *networktypes.NetworkChange) (bool, error) {
-	sequential := true
-	for index := r.changeIndex; index < change.Index; index++ {
-		priorChange, err := r.networkChanges.GetByIndex(index)
-		if err != nil {
-			return false, err
-		} else if priorChange != nil {
-			if priorChange.Status.State == changetypes.State_PENDING || priorChange.Status.State == changetypes.State_RUNNING {
-				if isIntersectingChange(change, priorChange) {
+	prevChange, err := r.networkChanges.GetPrev(change.Index)
+	if err != nil {
+		return false, err
+	}
+
+	for prevChange != nil {
```

Comment:
is there a risk this could whizz around at 100% CPU until it's `COMPLETE`? Could it do with a sleep or be event driven?

Reply:
There's no risk of that. At the end of the loop, the previous index is read:
```go
prevChange, err = r.networkChanges.GetPrev(prevChange.Index)
if err != nil {
	return false, err
}
```
Basically this is just iterating through the changes in reverse until the first intersecting change is found. So normally it will be a few reads.